src/pypromice/core/variables/precipitation.py

```diff
@@ -0,0 +1,101 @@
+__all__ = ["convert_to_rate", "filter"]
+
+import numpy as np
+import xarray as xr
+
+def filter(precip: xr.DataArray,
+           t: xr.DataArray,
+           p: xr.DataArray,
+           rh: xr.DataArray
+) -> xr.DataArray:
+    """Filter precipitation measurements where air temperature, pressure, or
+    relative humidity measurements are null values. This assumes that
+    air temperature, air pressure, relative humidity and precipitation
+    measurements are measured using the same instrument, e.g. a
+    lufft instrument.
+
+    Parameters
+    ----------
+    precip : xr.DataArray
+        Cumulative precipitation measurements
+    t : xr.DataArray
+        Air temperature measurements
+    p : xr.DataArray
+        Air pressure measurements
+    rh : xr.DataArray
+        Relative humidity measurements
+
+    Returns
+    -------
+    xr.DataArray
+        Filtered precipitation values
+    """
+    mask = (t.isnull() | p.isnull() | rh.isnull()) & (precip == 0)
+    return precip.where(~mask)
+
+def convert_to_rate(precip: xr.DataArray,
```

Comment:
To be more descriptive can we call it something like `convert_to_rate_and_correct_undercatch` ?